src/Ads/AdsRecommendationsService.php

```diff
@@ -0,0 +1,189 @@
+	 *
+	 * @param string $type Optional. Type of recommendation to retrieve. Currently supports only 'IMPROVE_PERFORMANCE_MAX_AD_STRENGTH'.
+	 * @param int    $id   Optional. Recommendation ID to filter by. Default 0.
+	 * @return array Array of recommendations.
+	 */
+	public function get_recommendations( string $type = '', int $id = 0 ): array {
+		/** @var AdsRecommendationsQuery $query */
+		$query = $this->container->get( AdsRecommendationsQuery::class );
+
+		if ( '' === $type ) {
+			$type = 'IMPROVE_PERFORMANCE_MAX_AD_STRENGTH';
+		}
+
+		// Filter by type if valid.
+		if ( $type === 'IMPROVE_PERFORMANCE_MAX_AD_STRENGTH' ) {
+			$query->where( 'recommendation_type', $type );
+		} else {
+			// If type is not valid, return an empty array.
+			return [];
+		}
+
+		if ( $id ) {
+			$query->where( 'recommendation_id', $id );
+		}
+
+		$result = $query->get_results();
+
+		$recommendations = [];
+
+		foreach ( $result as $item ) {
+			$recommendations[] = [
+				'id'              => (int) ( $item['recommendation_id'] ?? 0 ),
+				'type'            => $item['recommendation_type'] ?? '',
+				'resource_name'   => $item['recommendation_resource_name'] ?? '',
+				'campaign_id'     => (int) ( $item['recommendation_campaign_id'] ?? 0 ),
+				'campaign_name'   => $item['recommendation_campaign_name'] ?? '',
+				'campaign_status' => $item['recommendation_campaign_status'] ?? '',
+				'customer_id'     => (int) ( $item['recommendation_customer_id'] ?? 0 ),
+				'last_synced'     => isset( $item['recommendation_last_synced'] )
+					? gmdate( 'c', strtotime( $item['recommendation_last_synced'] ) )
+					: null,
+			];
+		}
+
+		return $recommendations;
+	}
+
+	/**
+	 * Retrieves recommendations from the Google Ads API.
+	 *
+	 * @param array $args Query arguments.
+	 * @return array Array of recommendations.
+	 *
+	 * @throws Exception If the recommendations data can't be retrieved.
+	 */
+	public function get_google_recommendations( $args ): array {
+		try {
+			$response = ( new GoogleAdsRecommendationsQuery( $args ) )
+			->set_client( $this->client, $this->options->get_ads_id() )
+			->get_results();
+
+			$result      = [];
+			$last_synced = gmdate( 'Y-m-d H:i:s' );
+			foreach ( $response->iterateAllElements() as $row ) {
+				if ( ! $row->hasRecommendation() ) {
+					continue;
+				}
+
+				$recommendation = $row->getRecommendation();
+
+				// Skip if recommendation is not valid.
+				if ( ! $recommendation instanceof Recommendation ) {
+					continue;
+				}
+
+				$campaign = $row->getCampaign();
+				$customer = $row->getCustomer();
+
+				$recommendation_id            = 0;
+				$recommendation_resource_name = method_exists( $recommendation, 'getResourceName' ) ? $recommendation->getResourceName() : '';
+				if ( $recommendation_resource_name ) {
+					$resource_name     = explode( '/', $recommendation_resource_name );
+					$recommendation_id = (int) end( $resource_name );
```

Comment:
@mukeshpanchal27 should we need to make sure we have both a resource name and can create the recommendation ID before accepting a recommendation?

I think we will always get a resource name and can generate the ID, but It might be worth implementing a check that we can generate these and if not skip this recommendation? 

Something along the lines of this?

```suggestion
				$recommendation_resource_name = method_exists( $recommendation, 'getResourceName' ) ? $recommendation->getResourceName() : '';
				
				if ( empty( $recommendation_resource_name ) ) {
					continue;
				}
				
				$resource_name     = explode( '/', $recommendation_resource_name );
				$recommendation_id = (int) end( $resource_name );
```

Reply:
The changes make sense. Update in https://github.com/woocommerce/google-listings-and-ads/pull/2991/commits/f3cf80179e426a207d75789ead3ea8c3aaffebd5